op-deployer/pkg/deployer/upgrade/upgrader.go

```diff
@@ -0,0 +1,159 @@
+package upgrade
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/ethereum-optimism/optimism/op-chain-ops/script"
+	"github.com/ethereum-optimism/optimism/op-deployer/pkg/deployer"
+	"github.com/ethereum-optimism/optimism/op-deployer/pkg/deployer/artifacts"
+	"github.com/ethereum-optimism/optimism/op-deployer/pkg/deployer/broadcaster"
+	"github.com/ethereum-optimism/optimism/op-deployer/pkg/env"
+	opcrypto "github.com/ethereum-optimism/optimism/op-service/crypto"
+	oplog "github.com/ethereum-optimism/optimism/op-service/log"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/urfave/cli/v2"
+)
+
+type Upgrader interface {
+	Upgrade(host *script.Host, input json.RawMessage) error
+	SupportsVersion(version string) bool
+	ArtifactsURL() string
+}
+
+func UpgradeCLI(upgrader Upgrader) func(*cli.Context) error {
+	return func(cliCtx *cli.Context) error {
+		logCfg := oplog.ReadCLIConfig(cliCtx)
+		lgr := oplog.NewLogger(oplog.AppOut(cliCtx), logCfg)
+		oplog.SetGlobalLogHandler(lgr.Handler())
+
+		ctx, cancel := context.WithCancel(cliCtx.Context)
+		defer cancel()
+
+		l1RPC := cliCtx.String(deployer.L1RPCURLFlag.Name)
+		if l1RPC == "" {
+			return fmt.Errorf("missing required flag: %s", deployer.L1RPCURLFlag.Name)
+		}
+		deploymentTarget, err := deployer.NewDeploymentTarget(cliCtx.String(deployer.DeploymentTargetFlag.Name))
+		if err != nil {
+			return fmt.Errorf("failed to parse deployment target: %w", err)
+		}
+
+		artifactsURL := upgrader.ArtifactsURL()
+		overrideArtifactsURL := cliCtx.String(OverrideArtifactsURLFlag.Name)
+		if overrideArtifactsURL != "" {
+			artifactsURL = overrideArtifactsURL
+		}
+		artifactsLocator, err := artifacts.NewLocatorFromURL(artifactsURL)
+		if err != nil {
+			return fmt.Errorf("failed to parse artifacts URL: %w", err)
+		}
+
+		rpcClient, err := rpc.Dial(l1RPC)
+		if err != nil {
+			return fmt.Errorf("failed to dial RPC %s: %w", l1RPC, err)
+		}
+		ethClient := ethclient.NewClient(rpcClient)
+
+		chainID, err := ethClient.ChainID(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to get chain ID: %w", err)
+		}
+
+		var bcaster broadcaster.Broadcaster
+		var depAddr common.Address
```

Comment:
Could save a few lines of code by initializing the value here to `depAddr := common.Address{'D'}`. Curious what is the significance of the 'D' value used though

Reply:
The `D` just means 'Deployer.'